workloads/virtual-machines/overcommit.md

```diff
@@ -0,0 +1,137 @@
+within a timespan specified via `--eviction-max-pod-grace-period`. The flag
+`--eviction-soft-grace-period` specifies for how long a soft eviction condition
+must be held before soft evictions are triggered.
+
+If set properly according to the demands of the VMIs, overcommiting should only
+lead to soft evictions from time to time for some VMIs. They may even get
+re-scheduled to the same node. This is fine because then they will start with
+less memory consumption again.
+
+### Configuring Hard Eviction Thresholds
+
+Limits set via `--eviction-hard` will lead to immediate eviction of
+VirtualMachineInstances or Pods. This stops VMIs without a grace period and is
+comparable with powerloss on a real computer.
+
+If this is hit, VMIs may from time to time simply be killed. The may be
+re-scheduled to the same node immediately again, since they start with less
+memory consumption again. This can be a simple option, if the memory threshold
+is only very seldom hit and the work performed by the VMIs is reproducible or
+it can be resumed from some checkpoints.
+
+### Requesting the right QoS Class for VirtualMachineInstances
+
+Different QoS classes get [assigned to Pods and
+VirtualMachineInstances](https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/#static-policy)
+based on the `requests.memory` and `limits.memory`. KubeVirt right now supports
+the QoS classes `Burstable` and `Guaranteed`. `Burstable` VMIs are evicted
+before `Guaranteed` VMIs.
+
+This allows creating two classes of VMIs: 
+
+* One type can have equal `requests.memory` and `limits.memory` set and therefore
+  gets the `Guaranteed` class assigned. This one will not get evicted and should
+  never run into memory issues, but is more demanding.
+* One type can have no `limits.memory` or a `limits.memory` which is greater than
+  `requests.memory` and therefore gets the `Burstable` class assigned. These VMIs
+  will be evicted first.
+
+### Setting `--system-reserved` and `--kubelet-reserved`
+
+It may be important to reserver some memory for other daemons (not DaemonSets)
+which are running on the same node (e.g. ssh, dhcp servers, ...). The
+reservation can be done with the `--system-reserved` switch. Furthter for the
+Kubelet and Docker a special flag called `--kubelet-reserved` exists.
+
+### Enabling KSM
+
+The KSM (Kernel same-page merging) daemon can be started on the node. Depending
+on its tuning parameters it can more or less aggressively try to merge
+identical pages between applications and VirtualMachineInstances. The more
+aggressive it is configured the more CPU it will use itself, so the memory
+overcommit advantages comes with a slight CPU performance hit.
+
+Possible tune options are on how often the pages should be scanned for merges
+and how much pages it should scan.
+
+### Enabling swap
+
+This will not work for Kubernetes `1.12` or later, since spilling over to swap,
```

Comment:
The second part of the sentence does not make much sense. I would provide a link and describe the relationship with QoS classes, because swap is a viable option for VMs as we discussed.

Reply:
Done.